beacon-chain/rpc/eth/v1/validator/validator.go

```diff
@@ -232,7 +235,51 @@ func (vs *Server) GetAggregateAttestation(ctx context.Context, req *v1.Aggregate
 
 // SubmitAggregateAndProofs verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
 func (vs *Server) SubmitAggregateAndProofs(ctx context.Context, req *v1.SubmitAggregateAndProofsRequest) (*emptypb.Empty, error) {
-	return nil, errors.New("Unimplemented")
+	ctx, span := trace.StartSpan(ctx, "validatorv1.GetAggregateAttestation")
+	defer span.End()
+
+	for _, agg := range req.Data {
+		if agg == nil || agg.Message == nil || agg.Message.Aggregate == nil || agg.Message.Aggregate.Data == nil {
+			return nil, status.Error(codes.InvalidArgument, "Signed aggregate request can't be nil")
+		}
+		sigLen := params.BeaconConfig().BLSSignatureLength
+		emptySig := make([]byte, sigLen)
+		if bytes.Equal(agg.Signature, emptySig) || bytes.Equal(agg.Message.SelectionProof, emptySig) {
```

Comment:
if you are checking signature integrity already, any reason to ignore `agg.Message.Aggregate.Signature`?